force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,7 +113,19 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
+        Integer result = 0;
+        Set <String> vowels = new Set <String>();
+        vowels.add('a');
+        vowels.add('e');
+        vowels.add('i');
+        vowels.add('o');
+        vowels.add('u');
+
+        for (String letter: str.split('')) {
+         if (vowels.contains(letter)){
+            result++;
+         }   
+        }
```

Comment:
This a a good start. But when you are checking for membership, "Is this character a member of the set of vowels? Using the Set datatype is much more performant.  My solution is blow as an example. 

```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```